panel/src/components/Forms/Login.vue

```diff
@@ -127,28 +147,30 @@ export default {
 		isResetForm() {
 			return this.codeMode === "password-reset" && this.form === "email";
 		},
+		submitText() {
+			const suffix = this.isLoading ? " …" : "";
+
+			if (this.isResetForm) {
+				return this.$t("login.reset") + suffix;
+			}
+
+			return this.$t("login") + suffix;
+		},
 		toggleText() {
 			return this.$t(
-				"login.toggleText." + this.codeMode + "." + this.formOpposite(this.form)
+				"login.toggleText." + this.codeMode + "." + this.alternateMode
 			);
 		}
 	},
 	methods: {
-		formOpposite(input) {
-			if (input === "email-password") {
-				return "email";
-			} else {
-				return "email-password";
-			}
-		},
 		async login() {
 			this.$emit("error", null);
 			this.isLoading = true;
 
 			// clear field data that is not needed for login
-			let user = Object.assign({}, this.user);
+			const user = Object.assign({}, this.user);
```

Comment:
I'm not entirely sure about it, but I think this should work as well and would feel closer to what we normally do …

```suggestion
			const user = {
			  ...this.user
			};
```